client/pubsub.go

```diff
@@ -3,61 +3,80 @@ package client
 import (
 	"context"
 	"encoding/json"
+	"log"
 
 	pb "github.com/dapr/go-sdk/dapr/proto/runtime/v1"
 	"github.com/pkg/errors"
 )
 
+// PublishEventOption is the type for the functional option.
+type PublishEventOption func(*pb.PublishEventRequest)
+
 // PublishEvent publishes data onto specific pubsub topic.
-func (c *GRPCClient) PublishEvent(ctx context.Context, pubsubName, topicName string, data []byte) error {
+func (c *GRPCClient) PublishEvent(ctx context.Context, pubsubName, topicName string, data interface{}, opts ...PublishEventOption) error {
 	if pubsubName == "" {
 		return errors.New("pubsubName name required")
 	}
 	if topicName == "" {
 		return errors.New("topic name required")
 	}
 
-	envelop := &pb.PublishEventRequest{
+	envelope := &pb.PublishEventRequest{
 		PubsubName: pubsubName,
 		Topic:      topicName,
-		Data:       data,
+	}
+	for _, opt := range opts {
+		opt(envelope)
+	}
+
+	if data != nil {
+		switch d := data.(type) {
+		case []byte:
+			envelope.Data = d
```

Comment:
Does a `DataContentType` of `application/octet-stream` make sense here?

Reply:
This is more tricky. As mentioned, the method tries to be backwards-compatible. Previously, the method only accepted a `[]byte` value as parameter (now it's an empty interface), and it didn't specify any content-type so by default that became `text/plain` (see my other comment).

I think that changing this to `application/octet-stream` would be a backwards-incompatible change that has the potential of breaking apps silently